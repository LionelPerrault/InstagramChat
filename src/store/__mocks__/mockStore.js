import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
