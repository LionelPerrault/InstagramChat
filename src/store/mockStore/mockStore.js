import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore
}

const middlewares = [
  promise(), thunk, logger
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
