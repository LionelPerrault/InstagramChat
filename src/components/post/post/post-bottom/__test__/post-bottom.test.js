import React from 'react'
import { create } from 'react-test-renderer'
import { Provider } from 'react-redux'
import mockStore from '../../../../../store/__mocks__/mockStore'
import PostBottom, { PurePostBottom } from '../post-bottom'
import { shallow } from 'enzyme'

describe('PostBottom Component', () => {
  const postDetails = {
    comments_count: 2,
    post_id: 44,
    when: 'feed',
    user: 7,
    comments: [],
  }

  it('should match snapshot', () => {
    const tree = create(
      <Provider store={mockStore}>
        <PostBottom postDetails={postDetails} />
      </Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should increments comments count', () => {
    const wrapper = shallow(<PurePostBottom postDetails={postDetails} />)
    expect(wrapper.state().comments_count).toBe(2)
    wrapper.instance().incrementComments()
    expect(wrapper.state().comments_count).toBe(3)
  })
})
