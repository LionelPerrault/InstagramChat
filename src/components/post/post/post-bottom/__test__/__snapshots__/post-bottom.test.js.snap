// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostBottom Component should match snapshot 1`] = `
<div>
  <Link
    className="p_comments"
    to="/post/44"
  >
    2 comments
  </Link>
  <div
    className="p_cit"
  >
    <div
      className="p_cit_img"
    >
      <img
        src="/users/24/avatar.jpg"
      />
    </div>
    <div
      className="p_cit_main"
    >
      <div
        className="p_cit_teaser"
        onClick={[Function]}
      >
        <span>
          Wanna comment?
        </span>
      </div>
      <div
        className="p_cit_tool"
      >
        <div>
          <span
            className="c_sticker"
            data-tip="Add sticker"
            onClick={[Function]}
          >
            <i
              className="material-icons"
            >
              face
            </i>
          </span>
          <div
            className="__react_component_tooltip place-top type-dark "
            data-id="tooltip"
          >
            
          </div>
        </div>
        <div>
          <form
            className="p_comment_form"
            encType="multipart/form-data"
          >
            <input
              accept="image/*"
              id="file_input"
              onChange={[Function]}
              type="file"
              value=""
            />
            <label
              className=""
              htmlFor="file_input"
            >
              <div
                data-tip="Attach a file"
              >
                <i
                  className="material-icons"
                >
                  attach_file
                </i>
              </div>
            </label>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
